solana-programs/anchor/audius-data/programs/audius-data/src/lib.rs

```diff
@@ -23,17 +23,36 @@ pub mod audius_data {
     pub fn init_admin(
         ctx: Context<Initialize>,
         authority: Pubkey,
+        authenticator: Pubkey,
         track_id_offset: u64,
         playlist_id_offset: u64,
     ) -> Result<()> {
         msg!("Audius::InitAdmin");
         let audius_admin = &mut ctx.accounts.admin;
         audius_admin.authority = authority;
+        audius_admin.authenticator = authenticator;
         audius_admin.track_id = track_id_offset;
         audius_admin.playlist_id = playlist_id_offset;
         Ok(())
     }
 
+    /// Verifies a user by asserting that the audius_admin's authenticator matches the signed authenticator account
+    pub fn verify_user(
+        ctx: Context<VerifyUser>
+    ) -> Result<()> {
+
+        // Question, do we need to validate the audius_admin account somehow?
```

Comment:
we do it by rederiving the PDA from seeds in `FollowUser` (admin instance + handle acct must exist) - since there is no back reference to the admin account in storage. open to changing that though if we think it is worth it.